datalad/ui/tests/test_dialog.py

```diff
@@ -173,25 +173,26 @@ def test_silent_question():
     # SilentConsoleLog must not be asked questions.
     # If it is asked, RuntimeError would be thrown with details to help
     # troubleshooting WTF is happening
-    from ..dialog import SilentConsoleLog
-    ui = SilentConsoleLog()
-    with assert_raises(RuntimeError) as cme:
-        ui.question("could you help me", title="Pretty please")
-    assert_in('question: could you help me. Title: Pretty please.', str(cme.value))
-
-    with assert_raises(RuntimeError) as cme:
-        ui.question("could you help me", title="Pretty please", choices=['secret1'], hidden=True)
-    assert_in('question: could you help me. Title: Pretty please.', str(cme.value))
-    assert_not_in('secret1', str(cme.value))
-    assert_in('not shown', str(cme.value))
-
-    # additional kwargs, no title, choices
-    with assert_raises(RuntimeError) as cme:
-        ui.question("q", choices=['secret1'])
-    assert_in('secret1', str(cme.value))
-
-
-@patch("datalad.log.is_interactive", lambda: False)
+    from ..dialog import SilentConsoleLog, NoDialogUnderAnnexUI
+    for k in [SilentConsoleLog, NoDialogUnderAnnexUI]:
```

Comment:
best to move that outside of the test as `@pytest.mark.parametrize`:
- minimizes `diff`
- fail in one doesn't taint the run of the other
- allows to run those two runs in parallel or selecting one particular one